proptools/compare.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proptools
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+)
+
+// DeepCompare is like reflect.deepEqual but it gives an explanation of the first difference that it finds
+func DeepCompare(aName string, aItem interface{}, bName string, bItem interface{}) (equal bool, difference string) {
+	return deepCompareGeneric(toReflection(aName, aItem), toReflection(bName, bItem), 0)
+}
+
+func same() (equal bool, difference string) {
+	return true, ""
+}
+
+func different(explanation string) (equal bool, difference string) {
+	return false, explanation
+}
+
+// end of public members
```

Comment:
This seems both unnecessary and misleading, since same() and different() are not public